2024/02/making-a-loading-spinner-with-tkinter.md

```diff
@@ -0,0 +1,303 @@
+---
+title: Making a Loading Spinner with tkinter
+date: 2024-02-27
+author: Matt Vollrath
+featured:
+  image_url: /blog/2024/02/making-a-loading-spinner-with-tkinter/spinner.webp
+tags:
+- linux
+- graphics
+- python
+- ubuntu
+- user-interface
+- visionport
+---
+
+When you need a loading spinner, you really need a loading spinner.  Interested in putting something on the screen without installing a pile of dependencies, I reached deep into the toolbox of the Python standard library, dug around a bit, and pulled out the [tkinter](https://docs.python.org/3/library/tkinter.html) module.
+
+The tkinter module is an interface to the venerable Tcl/Tk GUI toolkit, a cross-platform suite for creating user interfaces in the style of whatever operating system you run it on.  It's the only built-in GUI toolkit in Python, but there are many worthy alternatives available.
+
+Here I'll demonstrate how to make a loading spinnner with tkinter on Ubuntu 22.04.  It should work on any platform that runs Python, with some variations when setting up the system for it.
+
+### Prerequisites
+
+My vision for the loading spinner is some spinning dots and a logo, since this is such a convenient branding opportunity.  To accomplish this we'll be extending tkinter with Pillow's ImageTk capability, which can load a PNG with transparency.
+
+To produce that PNG with transparency, first we may need to rasterize an SVG file, because wise designers work in vectors.  This is made trivial by [inkscape](https://inkscape.org/), a free and complete vector graphics tool:
+
+`$ inkscape logo.svg -o logo.png`
+
+With the logo in hand, we can move on to setting up our dependencies.  Ubuntu's python3 distribution doesn't include tkinter by default, so we'll need to install it explicitly, along with Pillow's separate ImageTk support:
+
+`$ sudo apt install python3-tk python3-pil.imagetk`
+
+This may occupy up to 75MB, if Python was already installed.  This was still the smallest apt footprint of all of the Python GUI libraries in consideration.
```

Comment:
I'd be curious to hear which other libraries you tried, maybe just as a list in a sidenote.

Reply:
Added a mention here and a list at the end.